### PR TITLE
Fix/date picker input

### PR DESCRIPTION
**Pull request by betty-alagwu**
This PR is the first pass for issue #5452.

This PR allow users to type in the date in the `dateInput` and also select dates in the `datePicker`

Here's a video of how it works

https://github.com/highlight/highlight/assets/94234459/c92a1084-70c4-4a1d-97e2-77ef47a8411c

### CODE REVIEW

**Comment by julian-highlight** [comment]
@betty-alagwu can we maybe try to make the individual dates use the respective sizes? It seems like the width is much higher than the height of the individual items. In the designs they just differ by 2px. This is a small detail, which isn't to dramatic, but I'd love to keep it close to the designs to ensure we don't have future frustration with sizes.

File: packages/ui/src/components/DatePicker/DateInput.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+			name={name}
+			placeholder={placeholder}
+			type="input"
+			color={'n8'}
+			value={value}
+			style={{
+				boxSizing: 'border-box',
+				border: 'none',
+				background: 'none',
+				marginTop: '-5px',
```
Comment (Vadman97) [inline packages/ui/src/components/DatePicker/DateInput.tsx:34]: is this required for the font changes?

**Review by Vadman97 — APPROVED** [review]
code for parsing date in multiple formats looks great. regarding @julian-highlight 's comment, could we update the heights of the date grid?

**Comment by betty-alagwu** [comment]
Absolutely, will do that right away

File: packages/ui/src/components/DatePicker/DateInput.tsx (hunk 1 above)
Comment (betty-alagwu) [reply packages/ui/src/components/DatePicker/DateInput.tsx]: Not for the font change, It's required for the placement of the text and the borders of the container holding the text. I also made some changes to fix the fonts as well.
@Vadman97

**Pull request merged**